include/wil/registry_basic.h

```diff
@@ -791,247 +798,226 @@ namespace reg_view_details
-        return err_policy::HResult(HRESULT_FROM_WIN32(error));
-    }
-#endif
+            DWORD disposition = 0;
+            const auto error =
+                ::RegCreateKeyExW(::wil::reg::details::get_key(m_key), subKey, 0, nullptr, 0, ::wil::reg::details::get_access_flags(access), security_descriptor ? &security_attributes : nullptr, ::wil::out_param(hkey), &disposition);
+            return err_policy::HResult(HRESULT_FROM_WIN32(error));
+        }
 
-        ::wil::unique_hkey create_unique_key(_In_ PCWSTR subKey, ::wil::registry::key_access access = ::wil::registry::key_access::read, _In_opt_ PCWSTR security_descriptor = nullptr) const
+        ::wil::shared_hkey create_shared_key(_In_ PCWSTR subKey, ::wil::reg::key_access access = ::wil::reg::key_access::read, _In_opt_ PCWSTR security_descriptor = nullptr) const
         {
-            ::wil::unique_hkey local_key{};
+            ::wil::shared_hkey local_key{};
             create_key(subKey, local_key, access, security_descriptor);
             return local_key;
         }
-
-#if defined(__WIL_WINREG_STL)
-    ::wil::shared_hkey create_shared_key(_In_ PCWSTR subKey, ::wil::registry::key_access access = ::wil::registry::key_access::read, _In_opt_ PCWSTR security_descriptor = nullptr) const
-    {
-        ::wil::shared_hkey local_key{};
-        create_key(subKey, local_key, access, security_descriptor);
-        return local_key;
-    }
-#endif
+#endif // #if defined(__WIL_WINREG_STL)
 
         typename err_policy::result delete_key(_In_opt_ PCWSTR sub_key) WI_NOEXCEPT
         {
-            auto error = RegDeleteTreeW(::wil::registry::details::get_key(m_key), sub_key);
+            auto error = RegDeleteTreeW(::wil::reg::details::get_key(m_key), sub_key);
             if (error == ERROR_FILE_NOT_FOUND)
             {
                 error = ERROR_SUCCESS;
             }
             return err_policy::HResult(HRESULT_FROM_WIN32(error));
         }
 
-        typename err_policy::result delete_value(HKEY key, _In_opt_ PCWSTR value_name) WI_NOEXCEPT
+        typename err_policy::result delete_value(_In_opt_ PCWSTR value_name) WI_NOEXCEPT
         {
-            return err_policy::HResult(HRESULT_FROM_WIN32(RegDeleteValueW(key, value_name)));
+            return err_policy::HResult(HRESULT_FROM_WIN32(RegDeleteValueW(::wil::reg::details::get_key(m_key), value_name)));
         }
 
     private:
 
         template <typename R, typename get_err_policy = err_policy>
-        typename get_err_policy::result get_value_internal(_In_opt_ PCWSTR subkey, _In_opt_ PCWSTR value_name, R& return_value, DWORD type = ::wil::registry::reg_view_details::get_value_type<R>()) const
+        typename get_err_policy::result get_value_internal(_In_opt_ PCWSTR subkey, _In_opt_ PCWSTR value_name, R& return_value, DWORD type = ::wil::reg::reg_view_details::get_value_type<R>()) const
         {
             for (;;)

```

Comment:
Should we include a way of providing max-iterations for this? This loop can technically spin forever, right?